index.ts

```diff
@@ -2258,6 +2267,364 @@ async function getBranchDiffs(
+            content: content,
+          });
+        }
+      } else {
+        const fileContentUrl = new URL(
+          `${GITLAB_API_URL}/projects/${encodeURIComponent(effectiveProjectId)}/repository/files/${encodeURIComponent(diff.new_path)}/raw`
+        );
+        fileContentUrl.searchParams.append("ref", commit.id);
+
+        const fileContentResponse = await fetch(fileContentUrl.toString(), {
+          ...DEFAULT_FETCH_CONFIG,
+        });
+
+        if (fileContentResponse.ok) {
+          const content = await fileContentResponse.text();
+          actions.push({
+            action: "update",
+            file_path: diff.new_path,
+            content: content,
+          });
+        }
+      }
+    }
+
+    if (actions.length > 0) {
+      const commitUrl = new URL(
+        `${GITLAB_API_URL}/projects/${encodeURIComponent(effectiveProjectId)}/repository/commits`
+      );
+
+      const commitPayload: any = {
+        branch: tempBranchName,
+        commit_message: commit.title || commit.message || "Rebased commit",
+        actions: actions,
+      };
+
+      if (skipCi) {
+        commitPayload.commit_message += " [skip ci]";
+      }
+
+      const commitResponse = await fetch(commitUrl.toString(), {
+        ...DEFAULT_FETCH_CONFIG,
+        method: "POST",
+        body: JSON.stringify(commitPayload),
+      });
+
+      if (!commitResponse.ok) {
+        await deleteBranch(projectId, tempBranchName);
+        const errorBody = await commitResponse.text();
+        
+        if (commitResponse.status === 400 && errorBody.includes("conflict")) {
+          return {
+            success: false,
+            error: {
+              success: false,
+              message: "Rebase failed due to conflicts",
+              conflicts: commitDiffs.map((d: any) => d.new_path || d.old_path).filter(Boolean),
+            }
+          };
+        }
+        
+        throw new Error(`Failed to create commit: ${commitResponse.status} ${commitResponse.statusText}\n${errorBody}`);
+      }
+
+      const commitResult = await commitResponse.json() as any;
+      currentCommitSha = commitResult.id;
+    }
+  }
+
+  return { success: true, currentCommitSha };
+}
+
+/**
+ * Update the source branch to point to the new rebased commit
+ * 
+ * @param {string} effectiveProjectId - The effective project ID
+ * @param {string} sourceBranch - The source branch to update
+ * @param {string} newCommitSha - The new commit SHA to point to
+ */
+async function updateSourceBranch(
+  effectiveProjectId: string,
+  sourceBranch: string,
+  newCommitSha: string
+): Promise<void> {
+  const updateBranchUrl = new URL(
+    `${GITLAB_API_URL}/projects/${encodeURIComponent(effectiveProjectId)}/repository/branches/${encodeURIComponent(sourceBranch)}`
+  );
+
+  const updateBranchResponse = await fetch(updateBranchUrl.toString(), {
+    ...DEFAULT_FETCH_CONFIG,
+    method: "PUT",
+    body: JSON.stringify({
+      ref: newCommitSha,
+    }),
+  });
```

Comment:
https://docs.gitlab.com/api/branches/#get-single-repository-branch where is that PUT rest api? i could not find!